crd/apis/gcpfirewall/v1beta1/types.go

```diff
@@ -26,9 +26,8 @@ type Protocol string
 type CIDR string
 
 // +genclient
-// +genclient:nonNamespaced
 // +kubebuilder:object:root=true
-// +kubebuilder:resource:shortName=gf,scope=Cluster
+// +kubebuilder:resource:shortName=gf,scope=Namespace
```

Comment:
this is an interesting change, what happens if different namespaces have conflicting rules?

Reply:
Do you mean different namespaces with same names?.

It is the responsibility of the controllers to use a unique name (within the cluster) for the GCPFirwall CR. This is already being done as every controller creates a firewall rule name that is unique within the GCE project. 

The only reason we want to make this a namespaced resource is because we want to make use of ownerReference based garbage collection - https://kubernetes.io/docs/concepts/architecture/garbage-collection/#owners-dependents. With ownerreference, we can tie the lifecycle of a GCPFirewall CR to its parent resource. If we create a GCPFirewall CR for a k8s service, then GCPFirewall CR deletion is triggered automatically if the Service is deleted.